mesonpy/_util.py

```diff
@@ -71,3 +71,20 @@ def update(self, description: str) -> None:
     def __exit__(self, exc_type: Any, exc_value: Any, exc_tb: Any) -> None:
         if sys.stdout.isatty():
             print()
+
+
+def setup_windows_console() -> bool:
+    from ctypes import byref, windll  # type: ignore
+    from ctypes.wintypes import DWORD
+
+    STD_OUTPUT_HANDLE = -11
+    ENABLE_VIRTUAL_TERMINAL_PROCESSING = 0x04
+
+    kernel = windll.kernel32
+    stdout = kernel.GetStdHandle(STD_OUTPUT_HANDLE)
+    mode = DWORD()
+
+    if not kernel.GetConsoleMode(stdout, byref(mode)):
+        return False
+
+    return bool(kernel.SetConsoleMode(stdout, mode.value | ENABLE_VIRTUAL_TERMINAL_PROCESSING))
```

Comment:
The one difference with `mesonbuild/mlog.py` is that that adds `or os.environ.get('ANSICON')` (see https://github.com/adoxa/ansicon/blob/37f92009f6ca4938a34a1d7cecff41d331c7423c/readme.txt#L119 for the docs of that env var). It's not clear to me that that case is covered by the checks before calling `setup_windows_console` here - should it be added?

Reply:
I think it should be handled indeed. I had the impression that ansicon was some legacy thing, but I checked and the `ANSICON` environment variable is used by other terminal emulators on Windows. I'll add checking for the environment variable.